src/board.rs

```diff
@@ -72,16 +97,36 @@ impl Chessboard {
         }
     }
 
-    /// Create a new instance of a chessboard, based on a FEN string.
-    /// Forsyth–Edwards Notation Parser.
+    /// Creates a new instance of a chessboard based on a Forsyth–Edwards Notation (FEN) string.
     ///
     /// # Arguments
     ///
-    /// * `fen` - The FEN to be converted to a Chessboard.
+    /// - `fen`: The FEN string to be converted to a `Chessboard`.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` containing the resulting `Chessboard` with the position from the FEN string.
+    /// If the FEN string is invalid, an `Err` variant with an error message is returned.
+    ///
+    /// # Example
     ///
-    /// # Return
+    /// ```
+    /// use kn_o3::Chessboard;
     ///
-    /// Resulting chessboard with the position from the FEN.
+    /// let fen_string = "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1";
+    /// match Chessboard::from_string(fen_string) {
+    ///     Ok(chessboard) => {
+    ///         println!("Chessboard created from FEN:\n{:#?}", chessboard);
+    ///     },
+    ///     Err(error) => {
+    ///         println!("Error creating chessboard: {}", error);
+    ///     },
+    /// }
+    /// ```
```

Comment:
That's so cool that you can put markdown in docstrings. I wonder if other languages do that and maybe I've just never noticed.